metaflow/plugins/cards/card_cli.py

```diff
@@ -42,12 +43,68 @@ def open_in_browser(card_path):
     webbrowser.open(url)
 
 
+def resolve_task_from_pathspec(flow_name, pathspec):
+    """
+    resolves a task object for the pathspec query on the CLI.
+    Args:
+        flow_name : (str) : name of flow
+        pathspec (str) : can be `stepname` / `runid/stepname` / `runid/stepname/taskid`
+
+    Returns:
+        metaflow.Task | None
+    """
+    from metaflow import Flow, Step, Task
+    from metaflow.exception import MetaflowNotFound
+
+    # since pathspec can have many variations.
+    pthsplits = pathspec.split("/")
+    namespace(None)
+    task = None
+    run_id = None
+    resolving_from = "task_pathspec"
+    if len(pthsplits) == 1:
+        # This means stepname
+        resolving_from = "stepname"
+        latest_run = Flow(flow_name).latest_run
+        if latest_run is not None:
+            run_id = latest_run.pathspec
+            try:
+                task = latest_run[pathspec].task
+            except KeyError:
+                pass
+    elif len(pthsplits) == 2:
+        # This means runid/stepname
+        resolving_from = "step_pathspec"
+        try:
+            task = Step("/".join([flow_name, pathspec])).task
+            run_id = task.parent.parent.pathspec
+        except MetaflowNotFound as e:
+            pass
+    elif len(pthsplits) == 3:
+        # this means runid/stepname/taskid
+        resolving_from = "task_pathspec"
+        try:
+            task = Task("/".join([flow_name, pathspec]))
+            run_id = task.parent.parent.pathspec
+        except MetaflowNotFound as e:
+            pass
+    else:
+        # raise exception for invalid pathspec format
+        raise CommandException(
+            msg="The PATHSPEC argument should be of the form 'stepname' Or '<runid>/<stepname>' Or '<runid>/<stepname>/<taskid>'"
+        )
+
+    return task, run_id, resolving_from
+
+
 def resolve_card(
     ctx,
     pathspec,
     follow_resumed=True,
     hash=None,
     type=None,
+    card_id=None,
+    no_echo=False,
 ):
     """Resolves the card path based on the arguments provided. We allow identifier to be a pathspec or a id of card.
```

Comment:
TODO : Need to remove the comment : `We allow identifier to be a pathspec or a id of card.`